src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3006,6 +3012,10 @@ protected virtual void ProcessMethod (MethodDefinition method, in DependencyInfo
 			if (method.IsInstanceConstructor ()) {
 				MarkRequirementsForInstantiatedTypes (method.DeclaringType);
 				Tracer.AddDirectDependency (method.DeclaringType, new DependencyInfo (DependencyKind.InstantiatedByCtor, method), marked: false);
+			} else if (method.IsStaticConstructor () && Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method)) {
+				string formatString = SharedStrings.RequiresUnreferencedCodeOnStaticConstructorMessage;
+				string message = string.Format (formatString, method.GetDisplayName ());
+				_context.LogWarning (message, 2116, _scopeStack.CurrentScope.Origin, MessageSubCategory.TrimAnalysis);
```

Comment:
Please add the new ID to `DiagnosticId` and consider using it to retrieve the diagnostic string:

```suggestion
				_context.LogWarning (new DiagnosticString (DiagnosticId.RequiresUnreferencedCodeOnStaticConstructor), (int) DiagnosticId.RequiresUnreferencedCodeOnStaticConstructor, _scopeStack.CurrentScope.Origin, MessageSubCategory.TrimAnalysis);
```